config/config.go

```diff
@@ -110,6 +110,7 @@ type NodeConfig interface {
 type OverridableConfig interface {
 	NodeConfig
 	ForNode(nodeAddress primitives.NodeAddress, privateKey primitives.EcdsaSecp256K1PrivateKey) NodeConfig
+	MergeWithFileConfig(source string) (mutableNodeConfig, error)
 }
```

Comment:
there is something not so good having overridable returning a mutable, when mutable inherits from override ... looks bad ...

Reply:
I agree. its a smell. the problem here imho is that we have two interfaces with synonymous names: `Overridable` and `mutable`. whats the difference? only the public access...

but i think you're right... it should be consistent here..